src/lib/autodiscovery/autodiscovery.py

```diff
@@ -131,7 +215,28 @@ async def get_metric_descriptors(
             fetch_reslut.project_id
         )
 
-    return metric_per_project
+    return metric_per_project, resource_dimensions
+
+
+async def get_existing_metrics(
+    autodiscovery_resources: Dict[str, AutodiscoveryResourceLinking],
+) -> Dict[str, Set[str]]:
+    resources_to_metrics = {
+        resource_name: set() for resource_name in autodiscovery_resources.keys()
+    }
+
+    for resource_name, service_linking in autodiscovery_resources.items():
+        metric_name_list = []
+        if service_linking:
+            for service in service_linking.possible_service_linking:
+                for metric in service.metrics:
+                    metric_name_list.append(metric.google_metric)
+            for service in service_linking.disabled_services_for_resource:
+                for metric in service.metrics:
+                    metric_name_list.append(metric.google_metric)
+        resources_to_metrics[resource_name] = set(metric_name_list)
+
+    return resources_to_metrics
 
 
 async def run_autodiscovery(
```

Comment:
Could we add some short comments on main methods, like this one, to say what is being done? Like in fetch_ingest_lines_task(), in main.py. This is an extra, we should do this not only in this PR 😄